src/diffusers/pipelines/self_segmentation_stable_diffusion/self_segmentation_stable_diffusion.py

```diff
@@ -0,0 +1,968 @@
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether or not to return a [`~pipelines.stable_diffusion.StableDiffusionPipelineOutput`] instead of a
+                plain tuple.
+            callback (`Callable`, *optional*):
+                A function that will be called every `callback_steps` steps during inference. The function will be
+                called with the following arguments: `callback(step: int, timestep: int, latents: torch.FloatTensor)`.
+            callback_steps (`int`, *optional*, defaults to 1):
+                The frequency at which the `callback` function will be called. If not specified, the callback will be
+                called at every step.
+            cross_attention_kwargs (`dict`, *optional*):
+                A kwargs dictionary that if specified is passed along to the `AttentionProcessor` as defined under
+                `self.processor` in
+                [diffusers.cross_attention](https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/cross_attention.py).
+            guidance_rescale (`float`, *optional*, defaults to 0.7):
+                Guidance rescale factor proposed by [Common Diffusion Noise Schedules and Sample Steps are
+                Flawed](https://arxiv.org/pdf/2305.08891.pdf) `guidance_scale` is defined as `φ` in equation 16. of
+                [Common Diffusion Noise Schedules and Sample Steps are Flawed](https://arxiv.org/pdf/2305.08891.pdf).
+                Guidance rescale factor should fix overexposure when using zero terminal SNR.
+            segmentation_resolution (`int`, defaults to 32):
+                The resolution of the self-attention used for segmentation. The resultion should be on of (8, 16, 32,
+                64) which are the resolutions of the self-attention layers in the UNet.
+            segmentation_clustering_algorithm (`str`, defaults to "spectral"):
+                The clustering algorithm used for segmentation. Choose between "spectral" and "kmeans".
+            num_segments (`int`, defaults to 7):
+                The number of clusters used in the clustering algorithm when applying the segmentation. Better to put
+                larger number than the actual number of objects in the image, since we automatically merge clusters
+                that correspond to the same object.
+            seg_labels_indices_in_prompt (`List[int]`, defaults to None):
+                The indices of words in the prompt that corresponds to object labels. The labeling algorithm look at
+                the cross-attention maps of these indices. If None, nouns are automatically extracted from the prompt
+                and their indices are used. Automatically extracting the nouns requires nltk package.
+
+        Examples:
+
+        Returns:
+            [`~pipelines.stable_diffusion.BaseOutput`] or `tuple`: [`~pipelines.stable_diffusion.BaseOutput`] if
```

Comment:
Could we try to format this like 

https://github.com/huggingface/diffusers/blob/7a91ea6c2b53f94da930a61ed571364022b21044/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L648

?

Reply:
@sayakpaul I am not sure I understood you correctly, but if I did then I fixed it